integration-tests/smoke/keeper_test.go

```diff
@@ -171,33 +165,41 @@ func getKeeperSuite(
 
 		Describe("with Keeper job", func() {
 			if testToRun == BasicSmokeTest {
-				It("performs upkeep of a target contract, stops upon cancel", func() {
-					// Let upkeep be performed atleast once
-					Eventually(func(g Gomega) {
-						cnt, err := consumer.Counter(context.Background())
-						g.Expect(err).ShouldNot(HaveOccurred(), "Calling consumer's Counter shouldn't fail")
-						g.Expect(cnt.Int64()).Should(BeNumerically(">", int64(0)), "Expected consumer counter to be greater than 0, but got %d", cnt.Int64())
-						log.Info().Int64("Upkeep counter", cnt.Int64()).Msg("Upkeeps performed")
-					}, "1m", "1s").Should(Succeed())
-
-					// Now cancel the upkeep as registry owner, it should get immediately cancelled
+				It("monitors all the registered upkeeps perform and checks the cancel functionality for the last of them", func() {
+					for i := 0; i < len(upkeepIDs); i++ {
+						// Check if the upkeep is performing by analysing its counter and checking it is greater than 0
+						Eventually(func(g Gomega) {
+							counter, err := consumers[i].Counter(context.Background())
+							g.Expect(err).ShouldNot(HaveOccurred(), "Calling consumer's counter shouldn't fail")
+							g.Expect(counter.Int64()).Should(BeNumerically(">", int64(0)),
+								"Expected consumer counter to be greater than 0, but got %d", counter.Int64())
+							log.Info().Int64("Upkeep counter", counter.Int64()).Msg("Upkeeps performed")
+						}, "1m", "1s").Should(Succeed())
+					}
+
+					// For the upkeep which we registered last, we will also check if it is cancelled successfully.
+					// We don't do this for all the upkeeps because the cancellation checks take around 2 minutes.
```

Comment:
You can do all the cancelled checks within a single consistently, but not strictly required

Reply:
Hmm but the only thing that we want to consistently check is that the counter doesn't get increased once the upkeep is cancelled. So why would we put everything in one `Consistently` block?